version.go

```diff
@@ -390,25 +389,19 @@ func (v *Version) Original() string {
 	return v.original
 }
 
-// UnmarshalJSON implements JSON.Unmarshaler interface.
-func (v *Version) UnmarshalJSON(b []byte) error {
-	var s string
-	if err := json.Unmarshal(b, &s); err != nil {
-		return err
-	}
-	temp, err := NewVersion(s)
+// UnmarshalText implements encoding.TextUnmarshaler interface.
+func (v *Version) UnmarshalText(b []byte) error {
+	temp, err := NewVersion(string(b))
 	if err != nil {
 		return err
 	}
-	v.metadata = temp.metadata
-	v.pre = temp.pre
-	v.segments = temp.segments
-	v.si = temp.si
-	v.original = temp.original
+
+	*v = *temp
```

Comment:
This allows us to just copy the whole content of the struct without having to explicitly list out all fields.